DOC: sparse: Add doc for migration to sparray

dschult opened the pull request:
This is a building ground for migration to sparray docs. Comments and ideas welcome!!

Where should this live?  `doc/dev`?

Is `rst` the right format?

dschult committed:
Draft migration guide for converting from spmatrix to sparray [docs only]

dschult commented:
This is a Hackmd style migration document (soon to be converted to rst).
@stefanv and @perimosocordiae Are there obvious categories of changes I am leaving out?

stefanv commented on doc/source/dev/migration_to_sparray.rst:
```diff
@@ -0,0 +1,88 @@
+  - matrices are always 2D (rows x columns) even if there 1 row or column.
+    arrays can be 1D or 2D. Numpy scalars are used as 0D values where needed.  
+  - Operations change:
+    - matrices use operators * and @ for matrx multiplication, while arrays
```
```suggestion
    - matrices use operators * and @ for matrix multiplication, while arrays
```

stefanv commented on doc/source/dev/migration_to_sparray.rst:
```diff
@@ -0,0 +1,88 @@
+    arrays can be 1D or 2D. Numpy scalars are used as 0D values where needed.  
+  - Operations change:
+    - matrices use operators * and @ for matrx multiplication, while arrays
+      use * for elementwise multiplication and @ for matrix multiplication.
```
What do matrices use for scalar multiply? I.e. what do I look for in existing code that needs to change?

Same for power below.

stefanv reviewed (commented):
Nice, Dan, thank you!

In further shaping this, perhaps we could think about it in terms of user questions:

- In my code, what should change, from what to what?
- When writing new code, what should I watch out for, i.e. what is old vs new behavior.
- Are there any fundamental concepts / design choices that changed that I should keep in mind?

perimosocordiae commented:
Let's start with Markdown first (to get things rendering nicely), and then if needed we can convert to RST later.

perimosocordiae commented on doc/source/dev/migration_to_sparray.rst:
```diff
@@ -0,0 +1,88 @@
+
+The changeover from sparse matrices to sparse arrays involves changes
+in a few operations, changes in the output shape of some operations,
+and changes to the way some inpits are handled.
```
inpits -> inputs

perimosocordiae commented on doc/source/dev/migration_to_sparray.rst:
```diff
@@ -0,0 +1,88 @@
+### Migration from spmatrix to sparray
+
+The changeover from sparse matrices to sparse arrays involves changes
```
Let's start by saying that this mirrors the change from np.matrix to np.array, which hopefully we can find some useful pages on the web for. I wasn't able to find a canonical "migration from np.matrix to np.array" page though.

dschult replied:
I believe there never really was a migration from np.matrix to np.array.
Stefan and I were not able to find such previous documentation.

Still, I've added text to this point at the top.

perimosocordiae commented on doc/source/dev/migration_to_sparray.rst:
```diff
@@ -0,0 +1,88 @@
+def tril(A, k=0, format=None):
+def triu(A, k=0, format=None):
+
+# Names that should be rewritten to return sparse arrays
```
A table with these would be very nice.

ev-br committed:
TST: linalg: bump tolerance in TestEig::test_singular

Some assertions have atol/rtol configurable, and one assertion had them
hardcoded, and that was causing tolerance problems in a Debian build with
reference LAPACK.

closes https://github.com/scipy/scipy/issues/20911

lucascolley committed:
Merge pull request #20914 from ev-br/geneig_tol_tweak

larsoner committed:
BUG: sparse.csgraph.dijkstra: fix dtype and shape bugs (#20913)

j-bowhay committed:
ENH: stats: add array API support to combine_pvalues (#20900)

* ENH: stats: add array API support to combine_pvalues
